app/controllers/items_controller.rb

```diff
@@ -0,0 +1,84 @@
+class ItemsController < ApplicationController
+  def index
+    @items = Item.all
+  end
+
+  def new
+    @item = Item.new
+  end
+
+  def create
+    @item = Item.new(item_params)
+    @item.user = current_user
+    if @item.save
+      redirect_to item_path(@item)
+    else
+      flash[:notice] = "We could not save your treasure. Try again, matey!"
+      redirect_to new_item_path
+    end
+  end
+
+  def show
+    @item = Item.find_by(id: params[:id])
+    if @item && @item.user == current_user
+      @matches = @item.all_matches
+      @has_matches = !@matches.empty?
+    else
+      render :file => "#{Rails.root}/public/404.html", :status => 404
+    end
+  end
+
+  def available_items_show
+    if current_user
+      @swipe = Swipe.new
+      available_items = Item.available(current_user)
+      @item = available_items.shuffle.sample
+      if request.xhr?
+        if !available_items.empty?
+          render json: @item
+        else
+          render json:{message: "No More Items"}
+        end
+      else
+        @item
+      end
+    else
+      redirect_to login_path
+    end
+  end
+
+  def edit
+    @item = Item.find_by(id: params[:id])
+    if !@item
+      render :file => "#{Rails.root}/public/404.html", :status => 404
```

Comment:
@stevecass -- how do we write this so that you can only edit and see your own item? We've tried a few different things that don't work but we don't want to have to manually call the 404.